Common/include/toolboxes/C2DContainer.hpp

```diff
@@ -483,13 +538,31 @@ class C2DContainer :
   {
     for(size_t i=0; i<size(); ++i) m_data[i] = value;
   }
+
+  /*!
+   * \brief Get a scalar iterator to the inner dimension of the container.
+   */
+  template<size_t nCols>
+  FORCEINLINE CInnerIter innerIter(Index_t row) const noexcept
+  {
+    return CInnerIter(&m_data[IsRowMajor? row*nCols : row], IsRowMajor? 1 : this->rows());
+  }
+
+  /*!
+   * \brief Get a SIMD gather iterator to the inner dimension of the container.
+   */
+  template<size_t nCols, class T, size_t N>
+  FORCEINLINE CInnerIterGather<simd::Array<T,N> > innerIter(simd::Array<T,N> row) const noexcept
+  {
+    return CInnerIterGather<simd::Array<T,N> >(m_data, IsRowMajor? 1 : this->rows(), IsRowMajor? row*nCols : row);
+  }
```

Comment:
The only places where the "vector nature" of the SIMD type needs to be handled explicitly are when retrieving data from containers (C2DContainer and CVectorOfMatrix) and when putting it back in other containers (CSysVector and CSysMatrix).
Data is retrieved via iterators (see #789 for the reason, spoiler alert, it's for performance), the method to obtain the iterator is templated for the number of columns, i.e. the number of variables, because this is known at compile time in the new numerics (which in turn are templated for nDim). This is somewhat of a micro optimization, after benchmarking I might make it more general.